lib/matplotlib/transforms.py

```diff
@@ -139,11 +140,26 @@ def __setstate__(self, data_dict):
             k: weakref.ref(v, lambda _, pop=self._parents.pop, k=k: pop(k))
             for k, v in self._parents.items() if v is not None}
 
-    def __copy__(self, *args):
-        raise NotImplementedError(
-            "TransformNode instances can not be copied. "
-            "Consider using frozen() instead.")
-    __deepcopy__ = __copy__
+    def __copy__(self):
+        other = copy.copy(super())
+        # If `c = a + b; a1 = copy(a)`, then modifications to `a1` do not
+        # propagate back to `c`, i.e. we need to clear the parents of `a1`.
+        other._parents = {}
+        # If `c = a + b; c1 = copy(c)`, then modifications to `a` also need to
+        # be propagated to `c1`.
+        for key, val in vars(self).items():
+            if isinstance(val, TransformNode) and id(self) in val._parents:
+                other.set_children(val)  # val == getattr(other, key)
+        return other
+
+    def __deepcopy__(self, memo):
+        other = copy.deepcopy(super(), memo)
+        # If `c = a + b; a1 = deepcopy(a)`, then modifications to `a1` do not
+        # propagate back to `c`, i.e. we need to clear the parents of `a1`.
+        other._parents = {}
+        # If `c = a + b; c1 = deepcopy(c)`, this creates a separate tree
+        # (`c1 = a1 + b1`) so nothing needs to be done.
```

Comment:
OK, so with a deepcopy the tranform is frozen, or at least to the point where someone would need to dig out the children?  Does the `copy.deepcopy` above really copy the children over?